scss/_offcanvas.scss

```diff
@@ -0,0 +1,69 @@
+// stylelint-disable declaration-no-important
+.offcanvas {
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  z-index: $zindex-offcanvas;
+  width: $offcanvas-horizontal-width-mobile;
+  max-width: 100%;
+  padding: $offcanvas-padding-y $offcanvas-padding-x;
+  overflow-y: auto;
+  @include transition(transform $offcanvas-transition-duration ease-in-out);
+  transform: translateX(-100%);
+}
+
+.offcanvas-header {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: $offcanvas-padding-y;
+
+  .close {
+    padding: ($offcanvas-padding-y) ($offcanvas-padding-x / 2);
+    margin: (-$offcanvas-padding-y * 2) (-$offcanvas-padding-x) (-$offcanvas-padding-y) auto;
+  }
+}
+
+.offcanvas-right {
+  right: 0;
+  left: auto;
+  transform: translateX(100%);
+}
+
+.offcanvas-bottom {
+  top: auto;
+  right: 0;
+  width: 100%;
+  height: $offcanvas-vertical-height-mobile;
```

Comment:
I would remove the height, it makes more sense if the component scales to the content

Reply:
I will leaf a max height for safety reasons. ok?